### ISSUE TITLE
PLW2901 inconsistent with ior/iadd operations vx their method counterparts when in-place modiying

### ISSUE THREAD

**Issue by EkriirkE** [astral-sh/ruff#24986]
### Summary

The PLW2901 linter correctly does not warn if `.update()`'ing a dict-type iterator, but if you use `|=` it does warn.  Similarly `+=` a list vs `.append(), &c`.

I understand it might be hard for it to differentiate between cases where someone is incrementing an `int` with `+=`, but it should be easy enough to make exceptions when `|=` or `+=` is followed by an obvious dict/set/list

Basically; updating a collection of dicts/lists/sets in-place with operators vs methods

Or also warn when using (modification) methods for consistency.

### Version

2026-04-30 17:05:09.550 [info] Found Ruff 0.15.10 at ~\.vscode\extensions\charliermarsh.ruff-2026.40.0-win32-x64\bundled\libs\bin\ruff.exe

**Comment by MichaReiser** [comment]
Here a few examples:

```py
items = [{"a": 1}]

for item in items:
    item.update({"b": 2})  # no PLW2901

for item in items:
    item |= {"b": 2}  # PLW2901 today
```

```py
lists = [[1]]

for xs in lists:
    xs.append(2)  # no PLW2901

for xs in lists:
    xs += [2]  # PLW2901 today
```


```py
sets = [{1}]

for s in sets:
    s.update({2})  # no PLW2901

for s in sets:
    s |= {2}  # PLW2901 today
```

It would be interesting to see the ecosystem check for this.
